commands/build.py

```diff
@@ -66,11 +66,57 @@ def make(self):
         logger.info("building & installing {}: {}".format(self.name, " ".join(cmd)))
         subprocess.check_call(cmd, env=self.env)
 
+class acbuilder(object):
+    def __init__(self, name, modules_dir, build_dir, install_dir, host_prefix, make_verbose=False):
+        self.name = name
+        self.make_verbose = make_verbose
+        self.src_path = "{}/{}".format(modules_dir, name)
+        self.build_path = "{}/{}".format(build_dir, name)
+        self.install_path = install_dir
+        self.host_prefix = host_prefix
+        self.env = os.environ.copy()
+        self.env["STAGING_DIR"] = ""
+
+    def __str__(self):
+        return "'{}' builder configuration:\n\tsrc_path: {}\n\tbuild_path: {}\n\tinstall_path: {}".format(self.name, self.src_path, self.build_path, self.install_path)
+
+    def clean(self):
+        if os.path.exists(self.src_path):
+            logger.info("cleaning {}".format(self.name))
+            os.chdir(self.src_path)
+            os.system("make clean")
```

Comment:
this will clean only the build folder. What about the installed files? we don't want to have to do distclean and delete everything else.
I think that make uninstall should be supported for safec (used it myself).
For nng, you should use the install_manifest.txt (same line you removed for cmakebuilder) - 
`os.system("xargs rm < {}/install_manifest.txt".format(self.build_path))`